allennlp_rc/dataset_readers/bert_squad.py

```diff
@@ -0,0 +1,348 @@
+
+                    tok_to_orig_index = []
+                    orig_to_tok_index = []
+                    all_doc_tokens = []
+                    for (i, token) in enumerate(doc_tokens):
+                        orig_to_tok_index.append(len(all_doc_tokens))
+                        sub_tokens = self._tokenizer.tokenize(token)
+                        for sub_token in sub_tokens:
+                            tok_to_orig_index.append(i)
+                            all_doc_tokens.append(sub_token)
+
+                    tok_start_position = orig_to_tok_index[start_position]
+                    if end_position < len(doc_tokens) - 1:
+                        tok_end_position = orig_to_tok_index[end_position + 1] - 1
+                    else:
+                        tok_end_position = len(all_doc_tokens) - 1
+                    (tok_start_position, tok_end_position) = _improve_answer_span(
+                        all_doc_tokens, tok_start_position, tok_end_position, self._tokenizer,
+                        orig_answer_text)
+
+                    # The -3 accounts for [CLS], [SEP] and [SEP]
+                    max_tokens_for_doc = self.max_sequence_length - len(query_tokens) - 3
+
+                    # We can have documents that are longer than the maximum sequence length.
+                    # To deal with this we do a sliding window approach, where we take chunks
+                    # of the up to our max length with a stride of `doc_stride`.
+                    _DocSpan = namedtuple(  # pylint: disable=invalid-name
+                        "DocSpan", ["start", "length"])
+                    doc_spans = []
+                    start_offset = 0
+                    while start_offset < len(all_doc_tokens):
+                        length = len(all_doc_tokens) - start_offset
+                        if length > max_tokens_for_doc:
+                            length = max_tokens_for_doc
+                        doc_spans.append(_DocSpan(start=start_offset, length=length))
+                        if start_offset + length == len(all_doc_tokens):
+                            break
+                        start_offset += min(length, self.doc_stride)
+
+                    for (doc_span_index, doc_span) in enumerate(doc_spans):
+                        tokens = []
+                        token_to_orig_map = {}
+                        token_is_max_context = {}
+
+                        # Query
+                        for token in query_tokens:
+                            tokens.append(token)
+
+                        # SEP token
+                        tokens.append(Token(sep_token))
+
+                        # Paragraph
+                        paragraph_start_id = len(tokens)
+                        for i in range(doc_span.length):
+                            split_token_index = doc_span.start + i
+                            token_to_orig_map[len(tokens)] = tok_to_orig_index[split_token_index]
```

Comment:
`token_to_orig_map` is not read anywhere, just written to.

Reply:
And I think `tok_to_orig_index` is only here to set up `token_to_orig_map`, so it's not needed either.

Reply:
I have found the problem that accounts for the difference in score and will fix it ASAP.
In my fixed version, `token_to_orig_map ` is used.